app/controllers/admin/groups/users_controller.rb

```diff
@@ -0,0 +1,45 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Admin::Groups
+  class UsersController < ApplicationController
+    include ActionView::Helpers::TextHelper
+    before_action :require_can_edit_access_groups!
+    before_action :set_group
+
+    def create
+      # add to the access group any users passed through
+      user_ids = clean_params[:user_ids].select(&:present?).map(&:to_i)
+      AccessGroup.transaction do
+        User.where(id: user_ids).find_each do |user|
+          @group.add(user)
```

Comment:
Let's update `AccessGroup#add(user)` to take an array of users (or an individual user) with Array.wrap`

Reply:
Once this is done, you don't need the transaction, in the controller, but might want it in the add method.

Reply:
In the add method, rather than looping over the array of users, I am adding them simultaneously to the existing `@group.users`. This is analogous to the approach for roles.

Reply:
By working always with arrays we don't need any loops or transactions.